fw/hpack.c

```diff
@@ -3213,7 +3082,10 @@ tfw_hpack_encoder_index(TfwHPackETbl *__restrict tbl,
 	    && atomic64_read(&tbl->guard) < 0)
 		goto out;
 
-	res = tfw_hpack_rbtree_find(tbl, hdr, &node, &place);
+	tfw_http_hdr_split(hdr, &h_name, &h_val, spcolon);
```

Comment:
The patch still introduces the new `tfw_http_hdr_split()` call. My intention with https://github.com/tempesta-tech/tempesta/pull/1920/#discussion_r1271363420 comment was that we're OK with to leave too many calls of  `tfw_http_hdr_split()` for #1411, but the new code (this PR) should not make the situation worse. Comment https://github.com/tempesta-tech/tempesta/pull/1920/#discussion_r1263825337 proposed to move `tfw_http_hdr_split()` here (as it is now) AND remove the call from `tfw_http_hdr_split()`, so in terms of number of `tfw_http_hdr_split()` calls we won't make the code worse.

Reply:
Well, header name and value are needed for `tfw_hpack_rbtree_find` to work properly. Previously header parsing had been performed inside each call to [tfw_hpack_node_compare](https://github.com/tempesta-tech/tempesta/blob/5a8e13bbbbc2dc97f18a8178d268c5be09cfff7d/fw/hpack.c#L2285). In this PR among other things I've replaced that hand-written parsing **per each** rbtree node comparison with `tfw_hpack_rbtree_find` that is performed once per lookup (not once per comparison during lookup).
Now in other PR (probably in related to #1411) we need to avoid multiple calls to `tfw_hpack_rbtree_find` during same header processing. IMO this should be done once in [`tfw_hpack_encode`](https://github.com/tempesta-tech/tempesta/blob/5a8e13bbbbc2dc97f18a8178d268c5be09cfff7d/fw/hpack.c#L3760) and [`tfw_hpack_transform`](https://github.com/tempesta-tech/tempesta/blob/5a8e13bbbbc2dc97f18a8178d268c5be09cfff7d/fw/hpack.c#L3771C1-L3771C20).
If you'll trace down what's going on on execution path of these function you'll notice that there's [`tfw_hpack_encoder_index`](https://github.com/tempesta-tech/tempesta/blob/5a8e13bbbbc2dc97f18a8178d268c5be09cfff7d/fw/hpack.c#L3196) that is problematic to blindly transform to accept split headers only.